### PR TITLE
Fix CVE 2015-9284

### LINKED ISSUE #102: OAuth vulnerability (CVE-2015-9284)
Going to keep an issue open for tracking this. Looks like omniauth/omniauth#809 is tracking fixes for this, but once a fix is settled and/or tootsuite comes up with a fix, we should merge it and make a new release.

### PR DESCRIPTION
**Pull request by 1011X**
This PR is meant to fix #102 through the changes recommended by [this guide][1].

To my understanding, since Mastodon doesn't currently use any other authentication providers, this doesn't affect either of us. However, if we decide to add providers in the future, this fix ensures users stay safe.

Since this introduces a new dependency, I tried to update `Gemfile.lock`, but that led me down a rabbithole that ultimately came to a dead-end and I didn't know how to proceed, so for now I'll just give what I have.

[1]: https://github.com/omniauth/omniauth/wiki/Resolving-CVE-2015-9284

### CODE REVIEW

**Review by clarfonthey — DISMISSED** [review]

File: Gemfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ end
 
 gem 'net-ldap', '~> 0.10'
 gem 'omniauth-cas', '~> 1.1'
+gem 'omniauth-rails_csrf_protection'
```
Comment (jhaye) [inline Gemfile]: I have no experience with Ruby package management, but shouldn't we pin a specific version, just like in the linked guide?

Comment (1011X) [reply Gemfile]: I'll pin the latest version then.

**Comment by clarfonthey** [comment]
You forgot a comma between the name and the version; Ruby automatically concatenates strings without any operators between them like Python and C.

**Comment by 1011X** [comment]
yep, got it fixed

**Review by clarfonthey — APPROVED** [review]

**Comment by clarfonthey** [comment]
Double checked the guide and it does look like you did everything right.

**Review by jhaye — APPROVED** [review]

**Pull request merged**